src/rdkafka.h

```diff
@@ -6904,6 +6926,39 @@ rd_kafka_group_result_name(const rd_kafka_group_result_t *groupres);
 RD_EXPORT const rd_kafka_topic_partition_list_t *
 rd_kafka_group_result_partitions(const rd_kafka_group_result_t *groupres);
 
+/**
+ * @brief Topic Partition Result provides per-topic+partition operation result
+ *
+ */
+
+/**
+ * @returns the topic name from the topic partition result object.
+ * @remark lifetime of the returned string is the same as the \p
+ *         partition_result.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_topic(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @returns the partition number from the topic partition result object.
+ */
+RD_EXPORT int32_t rd_kafka_topic_partition_result_partition(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @returns the error code from the topic partition result object.
+ */
+RD_EXPORT rd_kafka_resp_err_t rd_kafka_topic_partition_result_error(
+    const rd_kafka_topic_partition_result_t *partition_result);
```

Comment:
We should return `error_t *` for new APIs that contains the error string too.
```suggestion
RD_EXPORT const rd_kafka_error_t *rd_kafka_topic_partition_result_error(
    const rd_kafka_topic_partition_result_t *partition_result);
```